src/mon/NVMeofGwMon.cc

```diff
@@ -0,0 +1,530 @@
+
+    dout(20) << "availability " <<  m->get_availability() << " GW : " << m->get_gw_id() <<
+        " osdmap_epoch " << m->get_last_osd_epoch() << " subsystems " << m->get_subsystems() << dendl;
+
+    NvmeGwId gw_id = m->get_gw_id();
+    NvmeGroupKey group_key = std::make_pair(m->get_gw_pool(),  m->get_gw_group());
+    gw_availability_t  avail = m->get_availability();
+    bool propose = false;
+    bool nonce_propose = false;
+    bool timer_propose = false;
+    bool gw_created = true;
+    NVMeofGwMap ack_map;
+    auto& group_gws = map.created_gws[group_key];
+    auto gw = group_gws.find(gw_id);
+    const BeaconSubsystems& sub = m->get_subsystems();
+
+    if (avail == gw_availability_t::GW_CREATED){
+        if (gw == group_gws.end()) {
+           gw_created = false;
+           dout(10) << "Warning: GW " << gw_id << " group_key " << group_key << " was not found in the  map.Created_gws "<< map.created_gws <<dendl;
+           goto set_propose;
+        }
+        else {
+            dout(4) << "GW  prepares the full startup " << gw_id << " GW availability: " << pending_map.created_gws[group_key][gw_id].availability << dendl;
+            if(pending_map.created_gws[group_key][gw_id].availability == gw_availability_t::GW_AVAILABLE){
+                dout(4) << "GW marked as Available in the NVmeofGwMon database, performed full startup - Force gw to exit!" << gw_id <<dendl;
+                avail = gw_availability_t::GW_UNAVAILABLE;
+                // Monitor performs Force Failover for this GW in process_gw_map_gw_down
+            }
+            else if(pending_map.created_gws[group_key][gw_id].performed_full_startup == false){
+                pending_map.created_gws[group_key][gw_id].performed_full_startup = true;
+                propose = true;
+                goto set_propose;
+            }
+        }
+    }
+    else { // gw already created
+        if (gw != group_gws.end()) // if GW reports Available but in monitor's database it is Unavailable
+                                   // it means it did not perform "exit" after failover was set by NVMeofGWMon
+           if( pending_map.created_gws[group_key][gw_id].availability == gw_availability_t::GW_UNAVAILABLE  &&
+               pending_map.created_gws[group_key][gw_id].performed_full_startup == false &&
+               avail == gw_availability_t::GW_AVAILABLE ) {
+               ack_map.created_gws[group_key][gw_id] = pending_map.created_gws[group_key][gw_id];
+               ack_map.epoch = map.epoch;
+               dout(1) << " Force gw to exit: Sending ack_map to GW: " << gw_id << dendl;
+               auto msg = make_message<MNVMeofGwMap>(ack_map);
+               mon.send_reply(op, msg.detach());
+               goto false_return;
+           }
+    }
+
+    // At this stage the gw has to be in the Created_gws
+    if(gw == group_gws.end()){
+        dout(1) << "Error : Administratively deleted GW sends beacon " << gw_id <<dendl;
```

Comment:
Why is this an error? Can't a GW which hasn't gotten the newest maps continue sending beacons for a duration?
You can log this if you want for debugging, but not at level 1.